scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook

```diff
@@ -0,0 +1,456 @@
+
+    data_diagnostic_carvel_versions 
+    data_diagnostic_carvel_status 
+    data_diagnostic_carvel_logs 
+
+    printf "\\n[INFO]Ending get_carvel_components_info\\n"
+    
+}
+
+data_diagnostic_carvel_versions()     {
+
+    set +e
+
+    printf "\\n[INFO] Looking for kapp-controller and secret-gen controller versions and saving it in a file.\\n"
+    kapp_version=$(kubectl describe deployment kapp-controller -n kapp-controller | grep "kapp-controller.carvel.dev/version:")
+    if [[ -z "${kapp_version}" ]]; then
+        printf "[ERROR] ${RED}kapp is not installed in the kapp-controller namespace (this is if you are using Tanzu RabbitMQ commercial version only) ${END}\\n"
+        exit -1
+    fi
+
+    secret_gen_controller_version=$(kubectl describe deploy secretgen-controller -n secretgen-controller | grep "secretgen-controller.carvel.dev/version")
+    if [[ -z "${secret_gen_controller_version}" ]]; then
+        printf "[ERROR] ${RED}secret-gen controller is not installed in the secretgen-controller namespace (this is if you are using Tanzu RabbitMQ commercial version only) ${END}\\n"
+        exit -1
+    fi
+
+    echo "kapp version:                              $kapp_version" > $OUTPUT_DIRECTORY/carvel/info/carvel_versions.info
+    echo "secretgen-controller version:              $secret_gen_controller_version" >> $OUTPUT_DIRECTORY/carvel/info/carvel_versions.info
+
+    printf "\\n[INFO] kapp-controller and secret-gen controller versions saved in $OUTPUT_DIRECTORY/carvel/info/carvel_versions.log\\n"
+}
+
+data_diagnostic_carvel_status()    {
+
+    set -e
+
+    # kapp status namespace
+    printf "\\n[INFO] Looking for kapp objects status and definitions in kapp-controller namespace and saving it to a file.\\n"
+    kubectl get all -n $KAPP_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/kapp_get_all_objects_status.info
+    kubectl describe replicaset -n $KAPP_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/kapp_get_all_objects_status.info
+    kubectl describe deploy -n $KAPP_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/kapp_get_all_objects_status.info
+    kubectl get deploy -n $KAPP_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/kapp-controller-definition.yaml
+
+
+    # secretgen-controller status namespace
+    printf "[INFO] Looking for secretgen-controller objects status and definitions in secretgen-controller namespace and saving it to files in $OUTPUT_DIRECTORY/carvel/info.\\n"
+    kubectl get all -n $SECRETGEN_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl describe replicaset -n $SECRETGEN_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl describe deploy -n $SECRETGEN_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl get deploy -n $SECRETGEN_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/secretgen-controller-definition.yaml
+
+    # secretgen-controller status namespace
```

Comment:
Should be packages namespace in the comment

Reply:
Done!